src/Storages/MergeTree/MergeTreeReaderStream.cpp

```diff
@@ -134,63 +134,28 @@ size_t MergeTreeReaderStream::getRightOffset(size_t right_mark_non_included)
     size_t result_right_offset;
     if (0 < right_mark_non_included && right_mark_non_included < marks_count)
     {
-        auto right_mark = marks_loader.getMark(right_mark_non_included);
-        result_right_offset = right_mark.offset_in_compressed_file;
-
-        bool need_to_check_marks_from_the_right = false;
-
-        /// If the end of range is inside the block, we will need to read it too.
-        if (right_mark.offset_in_decompressed_block > 0)
```

Comment:
It looks like the main change is the removing of `need_to_check_marks_from_the_right` flag. But it is not clear why this may may be a fix. Could you write some comment about it?